### DISCUSSION TITLE
Error logs

### DISCUSSION THREAD

**Discussion by cbude** [keycloakify/keycloakify#42] [Q&A] [answered]
Hi! 

I didn't want to create an issue since this isn't directly related to keycloakify. 

We all know about the errors appearing in the logs regarding the templates when running keycloak as described in your [README](https://github.com/InseeFrLab/keycloakify#about-the-errors-related-to-objecttojson-in-keycloak-logs).

I was wondering if your keycloak expert know how to quiet down/remove logging of this specific error type? I tried reading the docs but I can't seem to mute the specific module responsible for handling templates. The log becomes quite unreadable when all you can see are template errors.

**Comment by garronej** [comment]
Hi,  
Thank you for rising up the issue,  
I tried to address it but:

![image](https://user-images.githubusercontent.com/6702424/137716844-28bc31cf-2a6c-40cf-8067-ff95600860a3.png)

https://freemarker.apache.org/docs/ref_directive_attempt.html

The only option I would see would be to convince the Keycloak team to make a flag available to avoid logging FreeMarker errors to the console.  
Or maybe they could provide a ftl method that return a JSON formatted string representing the `kcContext`.

**Comment by garronej** [comment]
By lack of a better solution I renamed the `objectToJson` function to `objectToJson_please_ignore_errors`.  [code](https://github.com/InseeFrLab/keycloakify/blob/f2eadf54414cab2e6833d7add6d4b89a451f1ac4/src/bin/build-keycloak-theme/generateFtl/common.ftl#L2)

**Comment by garronej** [comment, marked as answer]
Fixed in `v4.2.14`
